### PR TITLE
🌱 Add namespace defaulting for ClusterClass ControlPlane MachineInfrastructure

### LINKED ISSUE #5089: ClusterClass webhooks does not apply default namespace for spec.controlPlane.machineInfrastructure.ref.namespace
**What steps did you take and what happened:**
While testing ClusterClass I got the following error

```
Error from server (Invalid): error when creating "clusterclass.yaml": admission webhook "validation.clusterclass.cluster.x-k8s.io" denied the request: ClusterClass.cluster.x-k8s.io "clusterclass1" is invalid: spec.controlPlane.machineInfrastructure.ref.namespace: Invalid value: "": must be 'default'
```

**What did you expect to happen:**
spec.controlPlane.machineInfrastructure.ref.namespace should be defaulted to metadata.namespace

**Environment:**

- Cluster-api version: tip of main

/kind bug

### PR DESCRIPTION
**Pull request by sbueringer**
Signed-off-by: Stefan Büringer buringerst@vmware.com

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #5089

### CODE REVIEW

File: api/v1alpha4/clusterclass_webhook.go [hunk 1]
```diff
@@ -49,9 +49,15 @@ func (in *ClusterClass) Default() {
...
 		in.Spec.Infrastructure.Ref.Namespace = in.Namespace
 	}
+
 	if in.Spec.ControlPlane.Ref != nil && len(in.Spec.ControlPlane.Ref.Namespace) == 0 {
 		in.Spec.ControlPlane.Ref.Namespace = in.Namespace
 	}
+	if in.Spec.ControlPlane.MachineInfrastructure != nil && in.Spec.ControlPlane.MachineInfrastructure.Ref != nil &&
+		len(in.Spec.ControlPlane.MachineInfrastructure.Ref.Namespace) == 0 {
+		in.Spec.ControlPlane.MachineInfrastructure.Ref.Namespace = in.Namespace
+	}
```
[the remark this answers is not in the document]
Comment (srm09) [reply api/v1alpha4/clusterclass_webhook.go:59]: Can be used for defaulting the namespace of the machineDeployment refs too.

[the remark this answers is not in the document]
Comment (sbueringer) [reply api/v1alpha4/clusterclass_webhook.go]: Thx, good idea. Fixed.

**Review by fabriziopandini — COMMENTED** [review]
/lgtm

**Review by vincepri — APPROVED** [review]
/approve

**Pull request merged**